core/services/bulletprooftxmanager/eth_broadcaster.go

```diff
@@ -129,8 +129,8 @@ func (eb *ethBroadcaster) monitorEthTxs() {
 
 			// It is safe to process separate keys concurrently
 			// NOTE: This design will block one key if another takes a really long time to execute
+			wg.Add(len(keys))
```

Comment:
It is effectively the same. I thought that the old way might be racey, now I've walked myself through it I've realised its not.

However it appears to be golang best practice to add all the routines to the waitgroup in one hit like this, since doing it the other way _can_ result in insidious races under certain conditions.